generators/app/templates/infrastructure/src/middleware/auth/auth.js

```diff
@@ -18,27 +18,23 @@ const jwtTokenValidation = (ctx, next) => {
     secret: jwksRsa.koaJwtSecret(client),
     issuer: IDENTITY_AUTHORITY,
     algorithms: ["RS256"],
-    key: 'jwtdata'
+    key: 'jwtdata',
+    tokenKey: 'token',
   });
   return validateJwtToken(ctx, next);
 }
 
 const jwtTokenUserIdentification = async (ctx, next) => {
-  const token = ctx.req.headers.authorization || "";
   let externalUser = {};
```

Comment:
You can better write this statement as follows:
```
this.externalUser = {
        id: ctx?.state?.jwtdata?.sub,
        role: ctx?.state?.jwtdata?.role,
      };
```
This way you don't need that ugly `if` anymore and you're safe of possible empty `jwtdata`. 
 If you don't want to have empty `id` and `role` properties but a whole empty object, I'd say:
```
const { jwtdata, token } = ctx?.state ?? {}
ctx.token = token
ctx.externalUser = jwtdata ? { id: jwtdata.sub, role: jwtdata.role } : {}
```